src/wallet/spend.cpp

```diff
@@ -706,9 +706,11 @@ static BResult<CreatedTransactionResult> CreateTransactionInternal(
         // Reserve a new key pair from key pool. If it fails, provide a dummy
         // destination in case we don't need change.
         CTxDestination dest;
-        bilingual_str dest_err;
-        if (!reservedest.GetReservedDestination(dest, true, dest_err)) {
-            error = _("Transaction needs a change address, but we can't generate it.") + Untranslated(" ") + dest_err;
+        auto op_dest = reservedest.GetReservedDestination(true);
+        if (op_dest.HasRes()) {
+            dest = op_dest.ReleaseObj();
+        } else {
+            error = _("Transaction needs a change address, but we can't generate it.") + Untranslated(" ") + op_dest.GetError();
         }
         scriptChange = GetScriptForDestination(dest);
```

Comment:
Thanks, done. Scrictly speaking moving the scriptChange assignment into the conditional is a bit more than a refactor (it could have been done also before the BResult change), but it also makes sense to not unnecessarily call `GetScriptForDestination` in the failure case.